kaggle_environments/envs/lux_ai_2022/test_agents/python/lux/factory.py

```diff
@@ -0,0 +1,62 @@
+import math
+from sys import stderr
+import numpy as np
+from dataclasses import dataclass
+if __package__ == "":
+    from lux.weather import get_weather_config
+    from lux.cargo import UnitCargo
+    from lux.config import EnvConfig
+else:
+    from .weather import get_weather_config
+    from .cargo import UnitCargo
+    from .config import EnvConfig
+@dataclass
+class Factory:
+    team_id: int
+    unit_id: str
+    strain_id: int
+    power: int
+    cargo: UnitCargo
+    pos: np.ndarray
+    # lichen tiles connected to this factory
+    # lichen_tiles: np.ndarray
+    env_cfg: EnvConfig
+
+
+    def build_heavy_metal_cost(self, game_state):
+        unit_cfg = self.env_cfg.ROBOTS["HEAVY"]
+        return unit_cfg.METAL_COST
+    def build_heavy_power_cost(self, game_state):
+        current_weather = game_state.weather_schedule[game_state.real_env_steps]
+        weather_cfg = get_weather_config(current_weather, self.env_cfg)
+        unit_cfg = self.env_cfg.ROBOTS["HEAVY"]
+        return math.ceil(unit_cfg.POWER_COST * weather_cfg["power_loss_factor"])
```

Comment:
I thought we were going to have factories immune to weather (due to their alternative power source?)

Reply:
Oh I figured cold snaps applied to all machinery? All machines need to warm up more? Also one less specific thing to memorize in addition to I think cold snap applying to everything being intuitive.

Reply:
Maybe you were thinking that in dust storms factories don't gain extra power (which is the case).

Reply:
yeah that's it!